components/portal/widgets/controllers.js

```diff
@@ -457,53 +459,55 @@ define(['angular'], function(angular) {
     };
 
     var resolveTemplateStatus = function(now, actionStart, actionEnd) {
-      var result = '';
+      if (!now || !actionStart || !actionEnd) {
+        throw new TypeError('now, actionStart, and actionEnd must all be set');
+      }
+
+      if (moment(actionStart).isAfter(actionEnd)) {
+        throw new RangeError('actionStart must be before actionEnd');
+      }
+
       var statuses = [
         {
           name: 'upcoming',
           check: function() {
-            return now < actionStart && now < actionEnd;
+            return moment(now).isBefore(actionStart);
           },
-        }, {
+        },
+        {
           name: 'ongoing',
           check: function() {
-            return now > actionStart && now < actionEnd;
+            return moment(now).isBetween(actionStart, actionEnd);
           },
-        }, {
+        },
+         {
           name: 'lastDay',
           check: function() {
-            return now > actionStart && now < actionEnd
+            return moment(now).isBetween(actionStart, actionEnd)
               && 1 === resolveDaysLeft(now, actionEnd);
           },
-        }, {
+        },
+        {
           name: 'ended',
           check: function() {
-            return now > actionStart && now > actionEnd;
+            return moment(now).isAfter(actionEnd);
           },
         },
       ];
 
-      if (now && actionStart && actionEnd) {
-        angular.forEach(statuses, function(status) {
-          if (status.check()) {
-            result = status.name;
-          }
-        });
-      }
+      var result = '';
+      // TODO: rewrite as Array.prototype.find when IE support is dropped
```

Comment:
note, this isn't a drop in replacement.  If you write it with `find` the order of the statuses array should be reversed, as 'lastDay' is a subset of 'ongoing', so it needs to be found first. It's currently written to fall through to more stringent classifications.

Reply:
Very true, luckily there is an `Array.prototype.reverse` function that can handle reorder nicely.
Or it could just be reversed in the code :wink: